operator-common/src/main/java/io/strimzi/operator/common/DefaultAdminClientProvider.java

```diff
@@ -23,34 +23,57 @@
     @Override
     public Admin createAdminClient(String hostname, Secret clusterCaCertSecret, Secret keyCertSecret, String keyCertName) {
```

Comment:
It seems to me that this method now copes with various cases where kets in secrets are absent etc. I guess that must imply that the `AdminClientProvider` contract is broader, so we should probably javadoc what the various cases are in the `AdminClientProvider`, right?

Reply:
@tombentley can you take a look now please?

Reply:
I think this should be documented in the interface.

Reply:
I documented on the specific "Default" implementation, because the behaviour of this method is defined by the implementation itself. I mean, we could have another implementation where if you pass Secrets as null, you got an exception because it's a sort of "TLS only" interface implementation. Documenting on the interface level we are saying that all the implementations should behave in the same way, or? Of course they have same interface but could have a different internal behavior.